### PR TITLE
Add size input to radio and checkbox inputs

### PR DESCRIPTION
**Pull request by elenagarrone**
# Description

Add `size` input to `radio` and `checkbox` inputs.

Also fixes an issue with the `commitlint` step in the CI not picking up the correct commits - caused by how GH checkout works.

## Requirements

Requirements discussed here: https://github.com/Legal-and-General/canopy/discussions/874

Radio `sm`:
Screenshot 2022-08-09 at 16 23 40

Radio `lg`:
Screenshot 2022-08-09 at 16 23 46

Checkbox `sm`:
Screenshot 2022-08-09 at 16 23 29

Checkbox `lg`:
Screenshot 2022-08-09 at 16 23 34

# Checklist:

- [x] The commit messages follow the [convention for this project](./blob/master/docs/CONTRIBUTING.md#conventional-commits)
- [x] I have provided an adequate amount of test coverage
- [ ] I have added the functionality to the [test app](./blob/master/docs/CONTRIBUTING.md#build-test-application)
- [x] I have provided a story in storybook to document the changes
- [x] I have provided documentation in the notes section of the story
- [ ] I have added any new public feature modules to public-api.ts

### CODE REVIEW

File: projects/canopy/src/lib/forms/radio/radio-button.component.scss [hunk 1]
```diff
@@ -83,3 +93,15 @@
     background-color: var(--form-error-border-color);
   }
 }
+
+.lg-radio-button__label--sm::after {
+  --_radio-inner-margin-horizontal: calc(var(--radio-inner-height) / 2);
```
Comment (elenagarrone) [inline projects/canopy/src/lib/forms/radio/radio-button.component.scss:98]: The approach of using the css variables like this and marking them as `private` is taken from a video I've watched during lightning learning last week: https://www.youtube.com/watch?v=ZuZizqDF4q8&t=2146s

**Review by pauleustice — APPROVED** [review]

**Review by pauleustice — DISMISSED** [review]

**Review by owensgit — COMMENTED** [review]
Nice one @elenagarrone , just a few questions from me

File: projects/canopy/src/lib/forms/radio/radio-button.component.scss [hunk 2]
```diff
@@ -38,8 +38,8 @@
   top: 0;
   bottom: 0;
   display: inline-block;
-  height: var(--radio-outer-height);
-  width: var(--radio-outer-width);
+  height: var(--_radio-outer-height);
+  width: var(--_radio-outer-width);
```
[the remark this answers is not in the document]
Comment (elenagarrone) [reply projects/canopy/src/lib/forms/radio/radio-button.component.scss:42]: I took inspiration from video mentioned in [this comment](https://github.com/Legal-and-General/canopy/pull/983#discussion_r941489944). The underscore variables are `private` and only defined within this file.

File: projects/canopy/src/lib/forms/radio/radio-button.component.spec.ts [hunk 3]
```diff
@@ -56,47 +58,45 @@ describe('LgRadioButtonComponent', () => {
...
-            provide: LgErrorStateMatcher,
-            useFactory: () => instance(errorStateMatcherMock),
-          },
-        ],
-      }).compileComponents();
+  beforeEach(waitForAsync(() => {
+    errorStateMatcherMock = mock(LgErrorStateMatcher);
+    radioGroupMock = mock(LgRadioGroupComponent);
+    when(radioGroupMock.name).thenReturn('color');
+    when(radioGroupMock.variant).thenReturn('segment');
```
[the remark this answers is not in the document]
Comment (elenagarrone) [reply projects/canopy/src/lib/forms/radio/radio-button.component.spec.ts:65]: Yes, I think once we decide what to do with all the apps we will create a ticket for Canopy too.

File: projects/canopy/src/lib/forms/radio/radio-button.component.spec.ts [hunk 4]
```diff
@@ -122,6 +122,27 @@ describe('LgRadioButtonComponent', () => {
...
+
+    when(radioGroupMock.variant).thenReturn('radio');
+    fixture = TestBed.createComponent(LgRadioButtonComponent);
+    component = fixture.componentInstance;
+    component.size = 'sm';
+    fixture.detectChanges();
+
+    expect(
+      fixture.debugElement.query(By.css('.lg-radio-button__label--sm')).nativeElement,
+    ).toBeDefined();
```
[the remark this answers is not in the document]
Comment (elenagarrone) [reply projects/canopy/src/lib/forms/radio/radio-button.component.spec.ts:136]: Updated 👍

File: projects/canopy/src/lib/forms/toggle/toggle.component.spec.ts [hunk 5]
```diff
@@ -189,6 +191,27 @@ describe('LgToggleComponent', () => {
...
+  it('adds the checkbox size based on the variant', () => {
+    component.variant = 'switch';
+
+    expect(fixture.debugElement.query(By.css('.lg-toggle__checkbox--sm'))).toBeNull();
+
+    component.variant = 'checkbox';
+    component.size = 'sm';
+    fixture.detectChanges();
+
+    expect(
```
[the remark this answers is not in the document]
Comment (elenagarrone) [reply projects/canopy/src/lib/forms/toggle/toggle.component.spec.ts:203]: Updated 👍

File: projects/canopy/src/lib/forms/radio/radio-button.component.scss (hunk 1 above)
Comment (owensgit) [reply projects/canopy/src/lib/forms/radio/radio-button.component.scss:98]: An interesting, sounds good. I was going to say that we should add this to our styling best practices documentation, but then I released that we don't have one - I think? Maybe we could benefit from a markdown file explaining how we handle SCSS, e.g. using BEM, use of CSS custom properties, etc - and also private CSS vars like this too. Bu that's just a side note really.

Comment (elenagarrone) [reply projects/canopy/src/lib/forms/radio/radio-button.component.scss:98]: That's a great idea 👍 I will make sure to create a ticket for it in our backlog.

**Review by pauleustice — DISMISSED** [review]

**Review by owensgit — APPROVED** [review]

**Review by pauleustice — APPROVED** [review]

**Pull request merged**